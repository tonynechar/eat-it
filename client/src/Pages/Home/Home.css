.categories-container {
  display: flex;
  overflow: scroll;
  -ms-overflow-style: none;
  scrollbar-width: none;
}

.categories-container::-webkit-scrollbar {
  display: none;
}

.categories-title {
  margin: 10px 16px;
}


.icon-container {
  min-height: 120px;
  min-width: 140px;
  border-radius: 6px;
  box-shadow: 0 3px 10px rgb(0 0 0 / 0.2);

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  margin: 6px;
  padding: 6px;

}

.icon-container .icon {
  font-size: 40px;
}

.icon-container h4 {
  padding: 12px 18px;
}
