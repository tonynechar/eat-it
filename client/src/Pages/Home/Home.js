import React, { useEffect, useState } from 'react';
import RestaurantList from '../../Components/RestaurantList/RestaurantList';
import apiService from '../../Services/ApiService';
import './Home.css';

function Home() {

  const [restaurants, setRestaurants] = useState([]);
  const [category, setCategory] = useState('');
  
  useEffect(() => {
    apiService.getRestaurants()
    .then(data => setRestaurants(data));
  }, []);

  const handleClick = (e, selectedCategory) => {
    e.preventDefault();
    setCategory(selectedCategory);
  }

  return (
    <div>
      <div>
        Search
      </div>
      <div>
        <h3 className='categories-title'>Restaurants</h3>
        <div className='categories-container'>
          <div className='icon-container' onClick={(e) => handleClick(e, '')}>
            <p className='icon'>🍽</p>
            <h4>All</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Pizza 🍕')}>
            <p className='icon'>🍕</p>
            <h4>Pizza</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Pasta 🍝')}>
            <p className='icon'>🍝</p>
            <h4>Pasta</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Burgers 🍔')}>
            <p className='icon'>🍔</p>
            <h4>Burgers</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Fast Food 🍟')}>
            <p className='icon'>🍟</p>
            <h4>Fast Food</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Healthy 🥗')}>
            <p className='icon'>🥗</p>
            <h4>Healthy</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Sushi 🍣')}>
            <p className='icon'>🍣</p>
            <h4>Sushi</h4>
          </div>
          <div className='icon-container' onClick={(e) => handleClick(e, 'Seafood 🐟')}>
            <p className='icon'>🐟</p>
            <h4>Seafood</h4>
          </div>
        </div>
      </div>
      <div>
        <RestaurantList restaurants={restaurants} category={category}/>
      </div>
    </div>
  )
}

export default Home