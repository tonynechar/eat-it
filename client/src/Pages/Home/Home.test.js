import { render, screen } from '@testing-library/react'
import Home from './Home';

// unit test
describe ('Home component', () => {
  // snapshot test
  test('Should match the snapshot', () => {
    const { container } = render(<Home/>);
    expect(container.firstChild).toMatchSnapshot();
  })
  // assertion test
  test('Should render the headings', () => {
    render(<Home></Home>);
    screen.getByText(/Restaurants/);
  })
})

// integration test


// end-to-end test