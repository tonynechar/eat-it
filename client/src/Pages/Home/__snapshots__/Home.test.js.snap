// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should match the snapshot 1`] = `
<div
  class="home-container"
>
  <div>
    <h3
      class="categories-title"
    >
      Restaurants
    </h3>
    <div
      class="categories-container"
    >
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍽
        </p>
        <h4>
          All
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍕
        </p>
        <h4>
          Pizza
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍝
        </p>
        <h4>
          Pasta
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍔
        </p>
        <h4>
          Burgers
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍟
        </p>
        <h4>
          Fast Food
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🥗
        </p>
        <h4>
          Healthy
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍣
        </p>
        <h4>
          Sushi
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🐟
        </p>
        <h4>
          Seafood
        </h4>
      </div>
    </div>
  </div>
  <div>
    <div
      class="restaurant-list"
    />
  </div>
</div>
`;

exports[`Home component should display "Restaurants" in the header 1`] = `
<div
  class="home-container"
>
  <div>
    <h3
      class="categories-title"
    >
      Restaurants
    </h3>
    <div
      class="categories-container"
    >
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍽
        </p>
        <h4>
          All
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍕
        </p>
        <h4>
          Pizza
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍝
        </p>
        <h4>
          Pasta
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍔
        </p>
        <h4>
          Burgers
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍟
        </p>
        <h4>
          Fast Food
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🥗
        </p>
        <h4>
          Healthy
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍣
        </p>
        <h4>
          Sushi
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🐟
        </p>
        <h4>
          Seafood
        </h4>
      </div>
    </div>
  </div>
  <div>
    <div
      class="restaurant-list"
    />
  </div>
</div>
`;

exports[`Home component should match the snapshot 1`] = `
<div
  class="home-container"
>
  <div>
    <h3
      class="categories-title"
    >
      Restaurants
    </h3>
    <div
      class="categories-container"
    >
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍽
        </p>
        <h4>
          All
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍕
        </p>
        <h4>
          Pizza
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍝
        </p>
        <h4>
          Pasta
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍔
        </p>
        <h4>
          Burgers
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍟
        </p>
        <h4>
          Fast Food
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🥗
        </p>
        <h4>
          Healthy
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🍣
        </p>
        <h4>
          Sushi
        </h4>
      </div>
      <div
        class="icon-container"
      >
        <p
          class="icon"
        >
          🐟
        </p>
        <h4>
          Seafood
        </h4>
      </div>
    </div>
  </div>
  <div>
    <div
      class="restaurant-list"
    />
  </div>
</div>
`;
